tools/setup/emoji/emoji_names.py

```diff
@@ -1,1580 +1,2055 @@
+    "1f1f3-1f1ea": {"canonical_name": "flag: Niger", "aliases": []},
+    "1f1f3-1f1eb": {"canonical_name": "flag: Norfolk Island", "aliases": []},
+    "1f1f3-1f1ec": {"canonical_name": "flag: Nigeria", "aliases": []},
+    "1f1f3-1f1ee": {"canonical_name": "flag: Nicaragua", "aliases": []},
+    "1f1f3-1f1f1": {"canonical_name": "flag: Netherlands", "aliases": []},
+    "1f1f3-1f1f4": {"canonical_name": "flag: Norway", "aliases": []},
+    "1f1f3-1f1f5": {"canonical_name": "flag: Nepal", "aliases": []},
+    "1f1f3-1f1f7": {"canonical_name": "flag: Nauru", "aliases": []},
+    "1f1f3-1f1fa": {"canonical_name": "flag: Niue", "aliases": []},
+    "1f1f3-1f1ff": {"canonical_name": "flag: New Zealand", "aliases": []},
+    "1f1f4-1f1f2": {"canonical_name": "flag: Oman", "aliases": []},
+    "1f1f5-1f1e6": {"canonical_name": "flag: Panama", "aliases": []},
+    "1f1f5-1f1ea": {"canonical_name": "flag: Peru", "aliases": []},
+    "1f1f5-1f1eb": {"canonical_name": "flag: French Polynesia", "aliases": []},
+    "1f1f5-1f1ec": {"canonical_name": "flag: Papua New Guinea", "aliases": []},
+    "1f1f5-1f1ed": {"canonical_name": "flag: Philippines", "aliases": []},
+    "1f1f5-1f1f0": {"canonical_name": "flag: Pakistan", "aliases": []},
+    "1f1f5-1f1f1": {"canonical_name": "flag: Poland", "aliases": []},
+    "1f1f5-1f1f2": {"canonical_name": "flag: St. Pierre & Miquelon", "aliases": []},
+    "1f1f5-1f1f3": {"canonical_name": "flag: Pitcairn Islands", "aliases": []},
+    "1f1f5-1f1f7": {"canonical_name": "flag: Puerto Rico", "aliases": []},
+    "1f1f5-1f1f8": {"canonical_name": "flag: Palestinian Territories", "aliases": []},
+    "1f1f5-1f1f9": {"canonical_name": "flag: Portugal", "aliases": []},
+    "1f1f5-1f1fc": {"canonical_name": "flag: Palau", "aliases": []},
+    "1f1f5-1f1fe": {"canonical_name": "flag: Paraguay", "aliases": []},
+    "1f1f6-1f1e6": {"canonical_name": "flag: Qatar", "aliases": []},
+    "1f1f7-1f1ea": {"canonical_name": "flag: R\u00e9union", "aliases": []},
+    "1f1f7-1f1f4": {"canonical_name": "flag: Romania", "aliases": []},
+    "1f1f7-1f1f8": {"canonical_name": "flag: Serbia", "aliases": []},
+    "1f1f7-1f1fa": {"canonical_name": "flag: Russia", "aliases": []},
+    "1f1f7-1f1fc": {"canonical_name": "flag: Rwanda", "aliases": []},
+    "1f1f8-1f1e6": {"canonical_name": "flag: Saudi Arabia", "aliases": []},
+    "1f1f8-1f1e7": {"canonical_name": "flag: Solomon Islands", "aliases": []},
+    "1f1f8-1f1e8": {"canonical_name": "flag: Seychelles", "aliases": []},
+    "1f1f8-1f1e9": {"canonical_name": "flag: Sudan", "aliases": []},
+    "1f1f8-1f1ea": {"canonical_name": "flag: Sweden", "aliases": []},
+    "1f1f8-1f1ec": {"canonical_name": "flag: Singapore", "aliases": []},
+    "1f1f8-1f1ed": {"canonical_name": "flag: St. Helena", "aliases": []},
+    "1f1f8-1f1ee": {"canonical_name": "flag: Slovenia", "aliases": []},
+    "1f1f8-1f1ef": {"canonical_name": "flag: Svalbard & Jan Mayen", "aliases": []},
```

Comment:
Similarly, one might have to delete the colons (I think you mentioned somewhere that searching for flag emojis in particular wasn't working in your testing -- perhaps this is why?).

Might also want to replace these `&` with `and`, and lowercase all the letters.

Reply:
There are some that work with colons, and I tried removing "flag:" and the flags still didn't work, which leaves me pretty stumped.

Lowercasing and replacing "&" with "and" (in the script) both sound good!

Reply:
Huh. I'll be curious what it turns out to be, then!

Reply:
It was that there can't be any uppercase letters in the canonical names! The string comparison always turns the search string into lowercase and compares to names/aliases.

Reply:
Ah! That makes sense, then.

So implicitly the search is assuming the names are all lowercase, while it tries to implement a case-insensitive search. It wouldn't be hard to fix it to not make that assumption, but also we probably want these colon-style names to be in all lowercase anyway: https://github.com/zulip/zulip/pull/22067#discussion_r896256356

Reply:
This suggests a possible bug in our handling of custom emoji. We should either (a) enforce that custom-emoji names are in lowercase, or (b) make the search work properly when they're not. It's possible we already do (a).

Reply:
so... I think we do enforce it, but we don't have good error messages haha.

<img width="738" alt="image" src="https://user-images.githubusercontent.com/5634097/176826922-64623bc9-e719-4b99-8bb5-736e35561d62.png">

Reply:
https://chat.zulip.org/#narrow/stream/137-feedback/topic/emoji.20validation